core/iwasm/common/wasm_runtime_common.c

```diff
@@ -863,6 +863,183 @@ wasm_runtime_call_wasm(WASMExecEnv *exec_env,
     return false;
 }
 
+uint32 parse_args_to_uint32_array(WASMType *type,
+                                uint32 num_args, wasm_val_t *args,
+                                uint32 *out_argv)
+{
+    int i, p;
+
+    for (i = 0, p = 0; i < num_args; i++) {
+        switch (args[i].kind) {
+            case WASM_I32:
+                *(int32 *)out_argv = args[i].of.i32;
+                out_argv++;
+                p++;
+                break;
+            case WASM_I64:
+                *(int64 *)out_argv = args[i].of.i64;
```

Comment:
Should not directly convert out_argv to int64 array, it might be not 8-byte aligned, which might cause access error/crash in arm system. Had better use union to copy the two bytes, ref to wasm_application_execute_func.